### PR TITLE
fix NFI when used from the truffle partial evaluator

### PR DESCRIPTION
**Pull request by rschatz**
The partial evaluator didn't run the normal Graal invocation plugins like a standard Java compilation. Since the NFI plugin can't do anything useful when the function is not yet constant, it needs to be re-run during partial evaluation.

This PR also contains a unit test demonstrating the problem. It also includes an update of the truffle version, since the unit test triggers the NPE of graalvm/truffle#72.

### CODE REVIEW

**Comment by rschatz** [comment]
@mrigger, this should the native call performance problems in sulong.

File: graal/com.oracle.graal.truffle.test/src/com/oracle/graal/truffle/test/InstrumentationPETestLanguage.java [hunk 1]
```diff
@@ -100,6 +99,7 @@ protected boolean isObjectOfLanguage(Object object) {
         return false;
     }
 
+    @SuppressWarnings("deprecation")
     @Override
     protected Visualizer getVisualizer() {
         return null;
```
Comment (woess) [inline graal/com.oracle.graal.truffle.test/src/com/oracle/graal/truffle/test/InstrumentationPETestLanguage.java]: we can remove this method when https://github.com/graalvm/truffle/pull/68 is merged.

**Comment by woess** [comment]
@rschatz Thanks for the fix and for adding a test. LGTM.

File: graal/com.oracle.graal.truffle.test/src/com/oracle/graal/truffle/test/InstrumentationPETestLanguage.java (hunk 1 above)
Comment (woess) [inline graal/com.oracle.graal.truffle.test/src/com/oracle/graal/truffle/test/InstrumentationPETestLanguage.java]: anyway, the instrumentation tests will be updated soon.

**Comment by rschatz** [comment]
It's failing one of the PE tests. Apparently some of the default graph builder plugins can't run during PE because the GraphBuilderContext implementation used during PE is not complete.

I'll look into how to fix that.

File: graal/com.oracle.graal.truffle.test/src/com/oracle/graal/truffle/test/InstrumentationPETestLanguage.java (hunk 1 above)
Comment (rschatz) [inline graal/com.oracle.graal.truffle.test/src/com/oracle/graal/truffle/test/InstrumentationPETestLanguage.java]: Ok. Looks like that's going to be integrated soon, so I'll wait for that change before integrating this...

**Comment by rschatz** [comment]
Fixed, and also updated include graalvm/truffle#68. Please review again.
/cc @thomaswue (because of the PE changes)

**Comment by rschatz** [comment]
The truffle update breaks some truffle unit tests. Due to a bug in mx this doesn't show up in the travis gate. I'm putting this on hold until both the mx and the truffle bug are fixed.

**Comment by rschatz** [comment]
Ok, next try ;)

This is basically the same content as before, but with the truffle update removed, since that was merged separately in the meantime.

**Pull request merged**